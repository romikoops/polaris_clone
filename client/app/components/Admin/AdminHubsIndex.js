import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styles from './Admin.scss';
import { AdminSearchableHubs } from './AdminSearchables';
import FileUploader from '../../components/FileUploader/FileUploader';
import { adminHubs as hubsTip } from '../../constants';
export class AdminHubsIndex extends Component {
    constructor(props) {
        super(props);
        this.state = {
        };
    }
    render() {
        const {theme, hubs, viewHub, adminDispatch} = this.props;
        const hubUrl = '/admin/hubs/process_csv';
        return(
            <div className="flex-100 layout-row layout-wrap layout-align-start-start">
                <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
                    <p className="flex-none">Upload Hubs Sheet</p>
                    <FileUploader
                        theme={theme}
                        url={hubUrl}
                        type="xlsx"
                        text="Hub .xlsx"
                    />
                </div>
                <AdminSearchableHubs
                    theme={theme}
                    hubs={hubs}
                    adminDispatch={adminDispatch}
                    sideScroll={false}
                    handleClick={viewHub}
                    seeAll={false}
                    icon="fa-info-circle"
                    tooltip={hubsTip.manage}
                />
            </div>
        );
    }
}
AdminHubsIndex.propTypes = {
    theme: PropTypes.object,
    hubs: PropTypes.array,
    clients: PropTypes.array,
    viewClient: PropTypes.func
};
