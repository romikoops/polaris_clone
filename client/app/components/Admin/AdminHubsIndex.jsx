import React from 'react'
import PropTypes from '../../prop-types'
import styles from './Admin.scss'
import { AdminSearchableHubs } from './AdminSearchables'
import FileUploader from '../../components/FileUploader/FileUploader'

export function AdminHubsIndex ({
  theme, hubs, viewHub, adminDispatch
}) {
  const hubUrl = '/admin/hubs/process_csv'
  return (
    <div className="flex-100 layout-row layout-wrap layout-align-start-start">
      <div className={`flex-100 layout-row layout-align-space-between-center ${styles.sec_upload}`}>
        <p className="flex-none">Upload Hubs Sheet</p>
        <FileUploader theme={theme} url={hubUrl} type="xlsx" text="Hub .xlsx" />
      </div>
      <AdminSearchableHubs
        theme={theme}
        hubs={hubs}
        adminDispatch={adminDispatch}
        sideScroll={false}
        handleClick={viewHub}
        seeAll={false}
      />
    </div>
  )
}

AdminHubsIndex.propTypes = {
  theme: PropTypes.theme,
  hubs: PropTypes.arrayOf(PropTypes.hub),
  viewHub: PropTypes.func.isRequired,
  adminDispatch: PropTypes.shape({
    getHub: PropTypes.func
  }).isRequired
}

AdminHubsIndex.defaultProps = {
  theme: null,
  hubs: []
}

export default AdminHubsIndex
