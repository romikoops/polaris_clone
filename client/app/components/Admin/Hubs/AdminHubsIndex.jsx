import React, { Component } from 'react'

import PropTypes from '../../../prop-types'
import styles from '../Admin.scss'
import FileUploader from '../../../components/FileUploader/FileUploader'

import { RoundButton } from '../../RoundButton/RoundButton'
import DocumentsDownloader from '../../../components/Documents/Downloader'

import SideOptionsBox from '../SideOptions/SideOptionsBox'
import CollapsingBar from '../../CollapsingBar/CollapsingBar'
import AdminHubsComp from './AdminHubsComp' // eslint-disable-line

export class AdminHubsIndex extends Component {
  constructor (props) {
    super(props)
    this.state = {
      expander: {}
    }
    // this.toggleExpander = this.toggleExpander.bind(this)
  }

  toggleExpander (key) {
    this.setState({
      expander: {
        ...this.state.expander,
        [key]: !this.state.expander[key]
      }
    })
  }

  render () {
    const { expander } = this.state
    const {
      theme, viewHub, toggleNewHub, documentDispatch
    } = this.props
    const hubUrl = '/admin/hubs/process_csv'
    const scUrl = '/admin/service_charges/process_csv'
    const newButton = (
      <div className="flex-none layout-row">
        <RoundButton
          theme={theme}
          size="small"
          text="New Hub"
          active
          handleNext={() => toggleNewHub()}
          iconClass="fa-plus"
        />
      </div>
    )
    if (!this.props.hubs) {
      return ''
    }

    const actionNodes = [<SideOptionsBox
      header="Data manager"
      content={(
        <div className="flex-100 layout-row layout-wrap layout-align-center-start">
          <CollapsingBar
            showArrow
            collapsed={!expander.upload}
            theme={theme}
            styleHeader={{ background: '#E0E0E0', color: '#4F4F4F' }}
            handleCollapser={() => this.toggleExpander('upload')}
            text="Upload Data"
            faClass="fa fa-cloud-upload"
            content={(
              <div>
                <div
                  className={`${
                    styles.action_section
                  } flex-100 layout-row layout-align-center-center layout-wrap`}
                >
                  <p className="flex-100 center">Upload Hubs Sheet</p>
                  <FileUploader
                    theme={theme}
                    url={hubUrl}
                    type="xlsx"
                    text="Hub .xlsx"
                    dispatchFn={documentDispatch.uploadHubs}
                  />
                </div>
                <div
                  className={`${
                    styles.action_section
                  } flex-100 layout-row layout-align-center-center layout-wrap`}
                >
                  <p className="flex-100 center">Upload Local Charges Sheet</p>
                  <FileUploader
                    theme={theme}
                    url={scUrl}
                    type="xlsx"
                    text="Hub .xlsx"
                    dispatchFn={documentDispatch.uploadLocalCharges}
                  />
                </div>
              </div>
            )}
          />
          <CollapsingBar
            showArrow
            collapsed={!expander.download}
            theme={theme}
            styleHeader={{ background: '#E0E0E0', color: '#4F4F4F' }}
            handleCollapser={() => this.toggleExpander('download')}
            text="Download Data"
            faClass="fa fa-cloud-download"
            content={(
              <div>
                <div
                  className={`${
                    styles.action_section
                  } flex-100 layout-row layout-wrap layout-align-center-center`}
                >
                  <p className="flex-100 center">Download Hubs Sheet</p>
                  <DocumentsDownloader theme={theme} target="hubs" />
                </div>
                <div
                  className={`${
                    styles.action_section
                  } flex-100 layout-row layout-wrap layout-align-center-center`}
                >
                  <p className="flex-100 center">Download Ocean Local Charges Sheet</p>
                  <DocumentsDownloader
                    theme={theme}
                    target="local_charges"
                    options={{ mot: 'ocean' }}
                  />
                </div>
                <div
                  className={`${
                    styles.action_section
                  } flex-100 layout-row layout-wrap layout-align-center-center`}
                >
                  <p className="flex-100 center">Download Air Local Charges Sheet</p>
                  <DocumentsDownloader
                    theme={theme}
                    target="local_charges"
                    options={{ mot: 'air' }}
                  />
                </div>
              </div>
            )}
          />
          <CollapsingBar
            showArrow
            collapsed={!expander.new}
            theme={theme}
            styleHeader={{ background: '#E0E0E0', color: '#4F4F4F' }}
            handleCollapser={() => this.toggleExpander('new')}
            text="Create New Hub"
            faClass="fa fa-plus-circle"
            content={(
              <div
                className={`${
                  styles.action_section
                } flex-100 layout-row layout-wrap layout-align-center-center`}
              >
                {newButton}
              </div>
            )}
          />
        </div>
      )}
    />]

    return (
      <div className="flex-100 layout-row layout-wrap layout-align-start-start extra_padding_left">
        <AdminHubsComp
          actionNodes={actionNodes}
          handleClick={viewHub}
        />
      </div>

    )
  }
}

AdminHubsIndex.propTypes = {
  theme: PropTypes.theme,
  hubs: PropTypes.arrayOf(PropTypes.hub),
  viewHub: PropTypes.func.isRequired,
  toggleNewHub: PropTypes.func.isRequired,
  documentDispatch: PropTypes.shape({
    closeViewer: PropTypes.func,
    uploadHubs: PropTypes.func
  }).isRequired
}

AdminHubsIndex.defaultProps = {
  theme: null,
  hubs: []
}

export default AdminHubsIndex
