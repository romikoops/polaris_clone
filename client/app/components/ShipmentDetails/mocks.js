// Will remove it later
// ============================================
import { get } from 'lodash'

export const trucking = { preCarriage: { truckType: '' }, onCarriage: { truckType: '' } }
export const id = 4606
export const selectedDay = '2019-01-24T10:00:00.000Z'

export const ShipmentDetailsAvailableRoutes = [{
  itineraryId: 2863,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'ocean',
  origin: {
    stopId: 4656, hubId: 3023, hubName: 'Gothenburg Port', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4657, hubId: 3027, hubName: 'Qingdao Port', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}, {
  itineraryId: 2958,
  itineraryName: 'Gothenburg - Qingdao',
  modeOfTransport: 'air',
  origin: {
    stopId: 4846, hubId: 3030, hubName: 'Gothenburg Airport', nexusId: 597, nexusName: 'Gothenburg', latitude: 57.694253, longitude: 11.854048, country: 'SE', truckTypes: ['default']
  },
  destination: {
    stopId: 4847, hubId: 3034, hubName: 'Qingdao Airport', nexusId: 601, nexusName: 'Qingdao', latitude: 36.083811, longitude: 120.323534, country: 'CN', truckTypes: []
  }
}]

export const ShipmentDetails = {
  availableRoutes: ShipmentDetailsAvailableRoutes,
  availableMots: ['ocean', 'air']
}

export const cargoItemTypes = [{
  id: 25, dimension_x: null, dimension_y: null, description: 'Pallet', area: null, created_at: '2018-06-27T17:28:28.431Z', updated_at: '2018-06-27T17:28:28.431Z', category: 'Pallet'
}, {
  id: 22, dimension_x: null, dimension_y: null, description: 'Carton', area: null, created_at: '2018-06-27T17:28:28.417Z', updated_at: '2018-06-27T17:28:28.417Z', category: 'Carton'
}, {
  id: 23, dimension_x: null, dimension_y: null, description: 'Crate', area: null, created_at: '2018-06-27T17:28:28.422Z', updated_at: '2018-06-27T17:28:28.422Z', category: 'Crate'
}, {
  id: 26, dimension_x: null, dimension_y: null, description: 'Bottle', area: null, created_at: '2018-06-27T17:28:28.436Z', updated_at: '2018-06-27T17:28:28.436Z', category: 'Bottle'
}, {
  id: 27, dimension_x: null, dimension_y: null, description: 'Stack', area: null, created_at: '2018-06-27T17:28:28.440Z', updated_at: '2018-06-27T17:28:28.440Z', category: 'Stack'
}, {
  id: 28, dimension_x: null, dimension_y: null, description: 'Drum', area: null, created_at: '2018-06-27T17:28:28.445Z', updated_at: '2018-06-27T17:28:28.445Z', category: 'Drum'
}, {
  id: 29, dimension_x: null, dimension_y: null, description: 'Skid', area: null, created_at: '2018-06-27T17:28:28.450Z', updated_at: '2018-06-27T17:28:28.450Z', category: 'Skid'
}, {
  id: 30, dimension_x: null, dimension_y: null, description: 'Barrel', area: null, created_at: '2018-06-27T17:28:28.454Z', updated_at: '2018-06-27T17:28:28.454Z', category: 'Barrel'
}]

export const maxDimensions = {
  general: {
    dimensionX: '590.0', dimensionY: '234.2', dimensionZ: '228.0', payloadInKg: '21770.0', chargeableWeight: '21770.0'
  },
  air: {
    dimensionX: '120.0', dimensionY: '100.0', dimensionZ: '150.0', payloadInKg: '1000.0', chargeableWeight: '1000.0'
  }
}

export const scope = {
  links: { about: '', legal: '' },
  terms: ['You verify that all the information provided above is true', 'You agree to the presented terms and conditions.', 'Demo is to discuss the validity of the presented prices with the product owners.'],
  fee_detail: 'key_and_name',
  closed_shop: false,
  has_customs: true,
  has_insurance: true,
  fixed_currency: false,
  dangerous_goods: false,
  cargo_info_level: 'hs_codes',
  carriage_options: { on_carriage: { export: 'optional', import: 'optional' }, pre_carriage: { export: 'optional', import: 'optional' } },
  detailed_billing: false,
  total_dimensions: true,
  consolidate_cargo: false,
  modes_of_transport: {
    air: { container: true, cargo_item: true }, rail: { container: true, cargo_item: true }, ocean: { container: true, cargo_item: true }, truck: { container: false, cargo_item: false }
  },
  show_beta_features: true,
  closed_registration: false,
  continuous_rounding: false,
  incoterm_info_level: 'text',
  non_stackable_goods: true,
  open_quotation_tool: false,
  customs_export_paper: false,
  fixed_exchange_rates: false,
  require_full_address: true,
  closed_quotation_tool: false
}

export const theme = {
  colors: {
    primary: '#0D5BA9', secondary: '#23802A', brightPrimary: '#2491FD', brightSecondary: '#25ED36'
  },
  logoWide: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_blue.png',
  logoLarge: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoSmall: 'https://assets.itsmycargo.com/assets/logos/logo_box.png',
  logoWhite: 'https://assets.itsmycargo.com/assets/logos/Logo_transparent_white.png',
  background: 'https://assets.itsmycargo.com/assets/images/cropped_banner_2.jpg'
}

export const origin = {
  latitude: 57.694253, longitude: 11.854048, nexusId: 597, nexusName: 'Gothenburg', country: 'SE'
}
export const destination = {
  latitude: 36.083811, longitude: 120.323534, nexusId: 601, nexusName: 'Qingdao', country: 'CN'
}
export const cargoItem = {
  payloadInKg: 11, totalVolume: 0, totalWeight: 0, dimensionX: 7, dimensionY: 7, dimensionZ: 8, quantity: 7, cargoItemTypeId: 23, dangerousGoods: false, stackable: false
}

export const cargoItemAggregated = {
  payloadInKg: 0, totalVolume: 122, totalWeight: 346, dimensionX: 0, dimensionY: 0, dimensionZ: 0, quantity: 1, cargoItemTypeId: '', dangerousGoods: false, stackable: true
}

export const cargoItemContainer = {
  sizeClass: 'highCube', quantity: 14, dangerousGoods: false, weight: 16
}

export const cargoUnits = [cargoItem]

export const identity = x => x

export const importedProps = {
  ShipmentDetails,
  aggregatedCargo: false,
  cargoItemTypes,
  cargoUnits,
  destination,
  direction: 'export',
  id,
  loadType: 'cargo_item',
  maxDimensions,
  onCarriage: false,
  onChangeCargoUnitCheckbox: identity,
  onChangeCargoUnitInput: identity,
  onChangeCargoUnitSelect: identity,
  onDeleteUnit: identity,
  onUpdateCargoUnit: identity,
  origin,
  preCarriage: false,
  scope,
  selectedDay,
  theme,
  toggleModal: identity,
  trucking
}

export function logPath (props, path) {
  console.log(path, get(props, path, 'NO SUCH PATH'))
}

export function log (props) {
  let holder = '{\n'
  const declarations = []
  Object.keys(props).forEach((key) => {
    const toLog = `const ${key} = ${JSON.stringify(props[key])}`
    if (toLog.length < 150) console.log(toLog)
    declarations.push(toLog)

    holder = `${holder}\n  ${key},`
  })
  holder = `${holder}\n}`
  console.log(declarations.join('\n'))
  // console.log(holder)
  // console.log(JSON.stringify(props, null, 2))
}
