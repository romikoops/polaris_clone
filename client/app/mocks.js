import { set, cloneDeep } from 'lodash'

/**
 * Checks if variable is an non-empty object
 *
 * True for `{a:1}`
 * Fase for `null, {}, [], [1,2,3]`
 */
const isObject = (x) => {
  const ok = x !== null && !Array.isArray(x) && typeof x === 'object'
  if (!ok) {
    return false
  }

  return Object.keys(x).length > 0
}
/* eslint-disable */
/**
 * Used in unit test to modify specific properties
 * with minimal lines of code
 *
 * You can specify `path` to be empty string
 * if the change affects more than one of object's branches
 */
export const change = (origin, pathRaw, rules) => {
  const willReturn = cloneDeep(origin)
  
  if (!isObject(rules)) {
    set(willReturn, pathRaw, rules)
    return willReturn
  }
  const path = pathRaw === '' ? '' : `${pathRaw}.`
  for (const ruleKey of Object.keys(rules)) {
    const rule = rules[ruleKey]
    if (!isObject(rule)) {
      set(
        willReturn,
        `${path}${ruleKey}`,
        rule
      )
      continue
    }
    Object.keys(rule).filter(subruleKey => !isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      set(
        willReturn,
        `${path}${ruleKey}.${subruleKey}`,
        subrule
      )
    })
    Object.keys(rule).filter(subruleKey => isObject(rule[subruleKey])).map((subruleKey) => {
      const subrule = rule[subruleKey]
      Object.keys(subrule).map((deepKey) => {
        const deep = rule[subruleKey][deepKey]
        set(
          willReturn,
          `${path}${ruleKey}.${subruleKey}.${deepKey}`,
          deep
        )
      })
    })
  }

  return willReturn
}
/* eslint-enable */
export const identity = input => input

export const theme = {
  colors: {
    primary: '#333',
    secondary: '#fafafa'
  }
}

export const user = {
  role: { name: 'shipper' },
  company_name: 'FOO_COMPANY',
  currency: 'EUR',
  email: 'foo@bar.baz',
  first_name: 'John',
  guest: false,
  last_name: 'Doe',
  phone: '6345789'
}

export const history = {
  push: identity
}

export const tenant = {
  data: {
    id: 123,
    scope: {
      modes_of_transport: {
        ocean: {
          OCEAN_LOAD_TYPE: true
        },
        air: {},
        truck: {},
        rail: {}
      }
    },
    theme,
    subdomain: 'foosubdomain'
  }
}

export const req = {
  schedule: [{}],
  total: 345,
  shipment: {}
}

const schedulesInShipmentData = [
  { hub_route_key: 'OCEAN_HUB_ROUTE_KEY', mode_of_transport: 'ocean' },
  { hub_route_key: 'AIR_HUB_ROUTE_KEY', mode_of_transport: 'air' },
  { hub_route_key: 'TRUCK_HUB_ROUTE_KEY', mode_of_transport: 'truck' },
  { hub_route_key: 'RAIL_HUB_ROUTE_KEY', mode_of_transport: 'rail' }
]

export const shipmentInShipmentData = {
  load_type: 'OCEAN_LOAD_TYPE',
  selected_offer: { cargo: {}, total: { value: 77 } },
  schedules_charges: {
    OCEAN_HUB_ROUTE_KEY: { total: 40 },
    AIR_HUB_ROUTE_KEY: { total: 200 },
    TRUCK_HUB_ROUTE_KEY: { total: 75 },
    RAIL_HUB_ROUTE_KEY: { total: 125 }
  },
  total_goods_value: {
    value: 15,
    currency: 'USD'
  },
  trucking: {
    pre_carriage: { trucking_time_in_seconds: 55 }
  },
  has_on_carriage: false,
  has_pre_carriage: false,
  notes: 'FOO_NOTES',
  cargo_notes: 'FOO_CARGO_NOTES',
  eori: 1234,
  incoterm_text: 'FOO_INCOTERM_TEXT',
  total_price: {
    value: 12,
    currency: 'USD'
  }
}

export const shipmentData = {
  contacts: [],
  shipment: shipmentInShipmentData,
  documents: [],
  cargoItems: [],
  containers: [],
  schedules: schedulesInShipmentData,
  locations: {
    startHub: 'FOO_START_HUB',
    endHub: 'FOO_END_HUB',
    destination: {
      street_number: 9,
      street: 'BAR_STREET',
      city: 'BAR_CITY',
      country: 'China',
      zip_code: 845321
    },
    origin: {
      street_number: 7,
      street: 'FOO_STREET',
      city: 'FOO_CITY',
      country: 'Germany',
      zip_code: 21177
    }
  }
}

export const match = {
  params: {}
}

export const shipments = {
  direction: 'export',
  finished: [],
  open: [],
  requested: []
}

export const shipment = {
  origin_hub: { name: 'FOO_ORIGIN_HUB' },
  destination_hub: { name: 'FOO_DESTINATION_HUB' },
  id: 654,
  status: 'FOO_STATUS',
  clientName: 'FOO_CLIENT_NAME',
  planned_etd: 789,
  imc_reference: 'FOO_IMC_REFERENCE',
  schedule_set: []
}

export const location = {
  primary: true,
  id: 246,
  street_number: 579,
  streetNumber: 579,
  geocodedAddress: 'FOO_GEOCODED_ADDRESS',
  street: 'FOO_STREET',
  city: 'FOO_CITY',
  zip_code: '22456',
  zipCode: '22456',
  country: 'Germany'
}

export const address = {
  id: 13579,
  street_number: '1',
  street: 'Uhlandweg',
  city: 'Hamburg',
  zip_code: '22848',
  country: 'Germany'
}

export const hub = {
  location,
  name: 'FOO_HUB_NAME'
}

export const client = {
  first_name: 'Sebastian',
  last_name: 'Muller',
  email: 'muller55@yahoo.de',
  company_name: 'Mehristweniger',
  password: 'pwd'
}

export const route = {
  id: 24680
}

export const vehicleType = {
  id: 827364,
  name: 'FOO_VEHICLE_TYPE_NAME',
  mode_of_transport: 'FOO_VEHICLE_TYPE_TRANSPORT'
}

export const schedule = {
  hub_route_key: 'FOO_SCHEDULE_ROUTE_KEY',
  id: 555777,
  eta: 441188
}

export const charge = {
  id: 654297
}

export const locations = {
  origin: {},
  destination: {}
}

class MapMock {
  constructor (x) {
    this.x = x
  }
  bindTo () {
    return this.x
  }
  setContent () {
    return this.x
  }
  addListener () {
    return this.x
  }
}

export const gMaps = {
  InfoWindow: MapMock,
  LatLngBounds: MapMock,
  Map: MapMock,
  MapTypeId: { ROADMAP: '' },
  Marker: MapMock,
  Point: MapMock,
  Size: MapMock,
  places: { Autocomplete: MapMock }
}
