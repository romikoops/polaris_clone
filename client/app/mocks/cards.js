export const firstCard = {
  name: 'FOO_NAME',
  img: 'FOO_IMG',
  url: 'FOO_URL',
  handleClick: x => x,
  options: {
    name: 'FOO_OPTIONS_NAME'
  }
}
export const secondCard = {
  name: 'BAR_NAME',
  img: 'BAR_IMG',
  url: 'BAR_URL',
  handleClick: x => x,
  options: {
    name: 'BAR_OPTIONS_NAME'
  }
}
export const cards = [
  firstCard,
  secondCard
]
