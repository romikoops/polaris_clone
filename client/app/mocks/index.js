export * from './address'
export * from './cards'
export * from './cargoItems'
export * from './client'
export * from './consignee'
export * from './contacts'
export * from './containers'
export * from './currencies'
export * from './dimensions'
export * from './documents'
export * from './feeHash'
export * from './functions'
export * from './gMaps'
export * from './group'
export * from './hub'
export * from './match'
export * from './notifyees'
export * from './offer'
export * from './results'
export * from './routes'
export * from './schedules'
export * from './scope'
export * from './shipmentData'
export * from './shipments'
export * from './shipper'
export * from './tenant'
export * from './theme'
export * from './user'
