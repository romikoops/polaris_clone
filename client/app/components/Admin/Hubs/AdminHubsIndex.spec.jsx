import * as React from 'react'
import { shallow } from 'enzyme'
import { theme, identity, hub, internalUser, user } from '../../../mocks/index'

import AdminHubsIndex from './AdminHubsIndex'

const propsBase = {
  theme,
  hubs: [hub],
  scope: {},
  viewHub: identity,
  toggleNewHub: identity,
  documentDispatch: {
    closeViewer: identity,
    uploadHubs: identity
  },
  user
}

test('shallow render', () => {
  expect(shallow(<AdminHubsIndex {...propsBase} />)).toMatchSnapshot()
})

test('hubs is falsy', () => {
  const props = {
    ...propsBase,
    hubs: null
  }
  expect(shallow(<AdminHubsIndex {...props} />)).toMatchSnapshot()
})

test('user.internal is truthy', () => {
  const props = {
    ...propsBase,
    user: internalUser
  }
  expect(shallow(<AdminHubsIndex {...props} />)).toMatchSnapshot()
})
