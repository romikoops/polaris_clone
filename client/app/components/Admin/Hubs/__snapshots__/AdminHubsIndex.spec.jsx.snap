// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hubs is falsy 1`] = `""`;

exports[`shallow render 1`] = `
<div
  className="flex-100 layout-row layout-wrap layout-align-start-start extra_padding_left"
>
  <Connect(AdminHubsComp)
    actionNodes={
      Array [
        <SideOptionsBox
          content={
            <div
              className="flex-100 layout-row layout-wrap layout-align-center-start"
            >
              
              <CollapsingBar
                collapsed={true}
                content={
                  <div>
                    <div
                      className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Download Hubs Sheet
                      </p>
                      <Connect(DocumentsDownloader)
                        t={[Function]}
                        target="hubs"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                      />
                    </div>
                    <div
                      className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Download Charge Definitions Sheet
                      </p>
                      <Connect(DocumentsDownloader)
                        t={[Function]}
                        target="charge_categories"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                      />
                    </div>
                    <div>
                      <div
                        className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                      >
                        <p
                          className="flex-100 center"
                        >
                          Download {{mot}} Local Charges Sheet
                        </p>
                        <Connect(DocumentsDownloader)
                          options={
                            Object {
                              "mot": "all",
                            }
                          }
                          t={[Function]}
                          target="local_charges"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                }
                contentHeader=""
                faClass="fa fa-cloud-download"
                handleCollapser={[Function]}
                headerWrapClasses=""
                hideIcon={false}
                mainWrapperStyle={Object {}}
                minHeight=""
                parentClass=""
                showArrow={true}
                showArrowSpacer={false}
                startCollapsed={false}
                styleHeader={
                  Object {
                    "background": "#E0E0E0",
                    "color": "#4F4F4F",
                  }
                }
                text="Download Data"
                theme={
                  Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  }
                }
                wrapperContentClasses=""
              />
              
            </div>
          }
          flexOptions="flex"
          header="Data manager"
        />,
      ]
    }
    handleClick={[Function]}
    t={[Function]}
  />
</div>
`;

exports[`user.internal is truthy 1`] = `
<div
  className="flex-100 layout-row layout-wrap layout-align-start-start extra_padding_left"
>
  <Connect(AdminHubsComp)
    actionNodes={
      Array [
        <SideOptionsBox
          content={
            <div
              className="flex-100 layout-row layout-wrap layout-align-center-start"
            >
              <CollapsingBar
                collapsed={true}
                content={
                  <div>
                    <div
                      className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Upload Hubs Sheet
                      </p>
                      <FileUploader
                        dispatchFn={[Function]}
                        formClasses=""
                        size="small"
                        square={false}
                        t={[Function]}
                        text="Hub .xlsx"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                        tooltip=""
                        type="xlsx"
                        uploadFn={null}
                        url="/admin/hubs/process_csv"
                      />
                    </div>
                    <div
                      className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Upload Charge Definitions Sheet
                      </p>
                      <FileUploader
                        dispatchFn={null}
                        formClasses=""
                        size="small"
                        square={false}
                        t={[Function]}
                        text="Charge Definitions .xlsx"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                        tooltip=""
                        type="xlsx"
                        uploadFn={null}
                        url="/admin/hubs/process_csv"
                      />
                    </div>
                    <div>
                      <div
                        className="action_section flex-100 layout-row layout-align-center-center layout-wrap"
                      >
                        <p
                          className="flex-100 center"
                        >
                          Upload {{mot}} Local Charges Sheet
                        </p>
                        <FileUploader
                          dispatchFn={[Function]}
                          formClasses=""
                          size="small"
                          square={false}
                          t={[Function]}
                          text="Hub .xlsx"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                          tooltip=""
                          type="xlsx"
                          uploadFn={null}
                          url="/admin/service_charges/process_csv"
                        />
                      </div>
                    </div>
                  </div>
                }
                contentHeader=""
                faClass="fa fa-cloud-upload"
                handleCollapser={[Function]}
                headerWrapClasses=""
                hideIcon={false}
                mainWrapperStyle={Object {}}
                minHeight=""
                parentClass=""
                showArrow={true}
                showArrowSpacer={false}
                startCollapsed={false}
                styleHeader={
                  Object {
                    "background": "#E0E0E0",
                    "color": "#4F4F4F",
                  }
                }
                text="Upload Data"
                theme={
                  Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  }
                }
                wrapperContentClasses=""
              />
              <CollapsingBar
                collapsed={true}
                content={
                  <div>
                    <div
                      className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Download Hubs Sheet
                      </p>
                      <Connect(DocumentsDownloader)
                        t={[Function]}
                        target="hubs"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                      />
                    </div>
                    <div
                      className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                    >
                      <p
                        className="flex-100 center"
                      >
                        Download Charge Definitions Sheet
                      </p>
                      <Connect(DocumentsDownloader)
                        t={[Function]}
                        target="charge_categories"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                      />
                    </div>
                    <div>
                      <div
                        className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                      >
                        <p
                          className="flex-100 center"
                        >
                          Download {{mot}} Local Charges Sheet
                        </p>
                        <Connect(DocumentsDownloader)
                          options={
                            Object {
                              "mot": "all",
                            }
                          }
                          t={[Function]}
                          target="local_charges"
                          theme={
                            Object {
                              "colors": Object {
                                "primary": "#333",
                                "secondary": "#fafafa",
                              },
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                }
                contentHeader=""
                faClass="fa fa-cloud-download"
                handleCollapser={[Function]}
                headerWrapClasses=""
                hideIcon={false}
                mainWrapperStyle={Object {}}
                minHeight=""
                parentClass=""
                showArrow={true}
                showArrowSpacer={false}
                startCollapsed={false}
                styleHeader={
                  Object {
                    "background": "#E0E0E0",
                    "color": "#4F4F4F",
                  }
                }
                text="Download Data"
                theme={
                  Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  }
                }
                wrapperContentClasses=""
              />
              <CollapsingBar
                collapsed={true}
                content={
                  <div
                    className="action_section flex-100 layout-row layout-wrap layout-align-center-center"
                  >
                    <div
                      className="flex-none layout-row"
                    >
                      <RoundButton
                        active={true}
                        back={false}
                        classNames=""
                        disabled={false}
                        flexContainer="100"
                        handleDisabled={null}
                        handleNext={[Function]}
                        icon=""
                        iconClass="fa-plus"
                        inverse={false}
                        size="small"
                        text="New Hub"
                        theme={
                          Object {
                            "colors": Object {
                              "primary": "#333",
                              "secondary": "#fafafa",
                            },
                          }
                        }
                      />
                    </div>
                  </div>
                }
                contentHeader=""
                faClass="fa fa-plus-circle"
                handleCollapser={[Function]}
                headerWrapClasses=""
                hideIcon={false}
                mainWrapperStyle={Object {}}
                minHeight=""
                parentClass=""
                showArrow={true}
                showArrowSpacer={false}
                startCollapsed={false}
                styleHeader={
                  Object {
                    "background": "#E0E0E0",
                    "color": "#4F4F4F",
                  }
                }
                text="Create New Hub"
                theme={
                  Object {
                    "colors": Object {
                      "primary": "#333",
                      "secondary": "#fafafa",
                    },
                  }
                }
                wrapperContentClasses=""
              />
            </div>
          }
          flexOptions="flex"
          header="Data manager"
        />,
      ]
    }
    handleClick={[Function]}
    t={[Function]}
  />
</div>
`;
